src/composite/MegaChart.js

```diff
@@ -52,41 +53,110 @@
+        if(this.showLegend()){
+            this._layout.setContent(this.legendPosition(), this._legend);
+        }
+        if(this.valueAxisTitle()){
+            this._layout.setContent("left", this._valueTitle.rotation(-90));
+        }
+        if(this.domainAxisTitle()){
+            this._layout.setContent("bottom", this._domainTitle);
+        }
         
         this.widget(this._layout);
     };
+    
     MegaChart.prototype.update = function (domNode, element) {
         Surface.prototype.update.apply(this, arguments);
         
-        if(this._layout !== null){
-            this._layout
-                .gutter(this.gutter())
-                .leftShrinkWrap(true)
-                .rightShrinkWrap(true)
-                .bottomShrinkWrap(true);
+        this._layout
+            .gutter(this.gutter())
+            .topShrinkWrap(true)
+            .leftShrinkWrap(true)
+            .rightShrinkWrap(true)
+            .bottomShrinkWrap(true);
+
+
+        this._chart
+                .columns(this.columns())
+                .data(this.data());
+
+        if(this._chart.chartType() !== this.chartType()){
+            this._chart.chartType(this.chartType());
+        }
+        
+        this.setLegend();
+        this._contentClasses = this.getContentClasses();
+        this.removeExtraLegends();
         
-            this._layout
-                .getContent("center")
-                    .columns(this.columns())
-                    .data(this.data());
-            
-            if(this._layout.getContent("center").chartType() !== this.chartType()){
-                this._layout.getContent("center").chartType(this.chartType());
+        this.setAxisTitles();
+        
+        if(this._legend !== null){
+            if(["top","bottom"].indexOf(this.legendPosition()) !== -1){
+                this._legend.orientation("horizontal");
+            } else {
+                this._legend.orientation("vertical");
             }
         }
-        if(this.showLegend()){
-            this._layout.setContent("right", new Legend().fixedSize(true).targetWidget(this._layout.getContent("center")));
-        } else {
-            this._layout.clearContent("right");
-        }
+        this._legend.dataType(this._chart.getChartDataType());
     };
+    
     MegaChart.prototype.exit = function (domNode, element) {
         Surface.prototype.exit.apply(this, arguments);
     };
+    
+    MegaChart.prototype.setLegend = function(){
+        if(this.showLegend() && (this._layout.getContent(this.legendPosition()) === null || this._layout.getContent(this.legendPosition()).classID() !== "other_Legend")){
+            this._legend = new Legend().fixedSize(true).targetWidget(this._layout.getContent("center"));
+            this._layout.setContent(this.legendPosition(), this._legend);
+        } else if (!this.showLegend()) {
+            this._layout.clearContent(this.legendPosition());
+        }
+    };
```

Comment:
Looks like you create a new legend every time it relocates?  If so you should remove the previous instance first (.target(null)) to avoid future leaks (probably not leaking in this case), but will need more work if the legend is to be shown on the dermatology page?